api/V1/Visits.ts

```diff
@@ -112,18 +112,21 @@ class DeviceVisits {
     queryOffset: number,
     complete: boolean = false
   ): Visit[] {
-    const visits = [];
+    const visits: Visit[] = [];
 
     const tracks = rec.Tracks;
     this.sortTracks(tracks);
-    for (const trackKey in tracks) {
-      const track = tracks[trackKey];
-      const visit = this.calculateTrackTagVisit(rec, track);
-      this.updateSummary(visit);
-      if (isVisit(visit)) {
-        visit.incomplete = !complete;
-        visit.queryOffset = queryOffset;
-        visits.push(visit);
+    for (const track of tracks) {
+      const event = this.calculateTrackTagEvent(rec, track);
+      if (event == null) {
+        continue;
+      }
+      this.updateSummary(event);
+      if (isVisit(event)) {
```

Comment:
Because you're using classes for Visit and VisitSummary, if you want to find out the type of one of these objects you can just do `if (event instanceof Visit) { ... }`, which is a bit more robust than checking for the existence of a particular property on the object.